mcdreforged/cli/cli_entry.py

```diff
@@ -157,3 +174,91 @@ def write(base_path: str, *, directory_only: bool):
 
 	writeln('Packed {} files/folders into "{}"'.format(file_counter, file_name))
 	writeln('Done')
+
+def init_plugin_workspace(path: str, pid: str, name: str, description: str, authors: str, link: str, resources: str, entrypoint: str, archive_name: str, *, quiet: bool = False):
+	writeln = print if not quiet else lambda *args, **kwargs: None
+	if quiet:
+		def ask(*args, default=None, **kwargs):
+			return default
+	else:
+		def ask(msg, *, default=None, skip=False):
+			r = input(('{} (enter to skip): ' if skip else '{}: ' if default is None else '{0} (default "{1}"): ').format(msg, default))
+			return default if len(r) == 0 else r
+
+	if path is None:
+		path = ask('Plugin work directory', default='.')
+	metafile = os.path.join(path, plugin_constant.PLUGIN_META_FILE)
+	if os.path.exists(metafile):
+		writeln('[ERROR] Meta file "{}" already exists'.format(metafile))
+		sys.exit(1)
+
+	if pid is None:
+		pid = ask('Id', default=os.path.basename(os.path.abspath(path)).lower()).strip()
+	if re.fullmatch(r'[0-9a-z_]{1,64}', pid) is None:
+		writeln('[ERROR] Plugin ID "{}" is invalid (match express: [0-9a-z_]{{1,64}})'.format(pid))
+		sys.exit(1)
+
+	if name is None:
+		name = ask('Name', default=pid).strip()
+
+	if description is None:
+		description = ask('Description', default='This is a plugin for MCDR').strip()
+
+	if authors is None:
+		authors = ask("Author(s), split with ':'")
+
+	if link is None:
+		link = ask('Main page link', skip=True)
+
+	if resources is None:
+		resources = ask("Resource(s), split with ':'", skip=True)
+
+	if entrypoint is None:
+		entrypoint = ask('Entry point', skip=True)
+
+	if entrypoint is not None and entrypoint != pid and not entrypoint.startswith(pid + '.'):
+		writeln('Invalid entry point "{0}" for plugin id "{1}"'.format(entrypoint, pid))
+		sys.exit(1)
+
+	if archive_name is None:
+		archive_name = ask('Archive name', skip=True)
+
+	metadata = {
+		'id': pid,
+		'version': '1.0.0',
+		'name': name,
+		'description': description,
+		'dependencies': {
+			'mcdreforged': '>=2.0.0'
+		}
+	}
+	if entrypoint is not None:
+		metadata['entrypoint'] = entrypoint
+	if authors is not None:
+		metadata['author'] = list(set((a.strip() for a in authors.split(':'))))
+	if link is not None:
+		metadata['link'] = link
+	if resources is not None:
+		metadata['resources'] = list(set(resources.split(':')))
+	if archive_name is not None:
+		metadata['archive_name'] = archive_name
+
+	with open(metafile, 'w') as fd:
+		json.dump(metadata, fd, indent=4)
```

Comment:
maybe adding an `encoding='utf8'` would be nicer? so Chinese windows user won't create a gbk encoding meta file